jni/CMakeLists.txt

```diff
@@ -28,12 +28,25 @@ cmake_minimum_required(VERSION 3.17)
 project(KNNPlugin_JNI)
 
 # ---------------------------------- SETUP ----------------------------------
-# Target library to be compiled
-set(TARGET_LIB opensearchknn)
+# Target libraries to be compiled
+set(TARGET_LIB_COMMON opensearchknn_common)  # Shared library with common utilities
+set(TARGET_LIB_NMSLIB opensearchknn_nmslib)  # nmslib JNI
+set(TARGET_LIB_FAISS opensearchknn_faiss)    # faiss JNI
+set(TARGET_LIBS "")  # Libs to be installed
 
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
+option(CONFIG_FAISS "Configure faiss library build when this is on")
+option(CONFIG_NMSLIB "Configure nmslib library build when this is on")
+option(CONFIG_TEST "Configure tests when this is on")
+
+if (${CONFIG_FAISS} STREQUAL OFF AND ${CONFIG_NMSLIB} STREQUAL OFF AND ${CONFIG_TEST} STREQUAL OFF)
+    set(CONFIG_ALL ON)
+else()
+    set(CONFIG_ALL OFF)
+endif ()
```

Comment:
```suggestion
if (${CONFIG_FAISS} STREQUAL ON OR ${CONFIG_NMSLIB} STREQUAL ON OR ${CONFIG_TEST} STREQUAL ON)
    set(CONFIG_ALL OFF)
endif ()
```

Reply:
But then if they are all on, config_all will be set to off.

Reply:
Good catch!